dataikuapi/dss/notebook.py

```diff
@@ -27,14 +28,19 @@ def unload(self, session_id=None):
                 raise Exception("Several sessions of the notebook are running, choose one")
             else:
                 session_id = state['activeSessions'][0].get('sessionId', None)
-        return self.client._perform_json("DELETE", "/projects/%s/notebooks/" % self.project_key, params={'notebookName' : self.notebook_name, 'sessionId' : session_id})
+        return self.client._perform_json("DELETE",
+                                         "/projects/%s/jupyter-notebooks/%s/sessions/%s" % (self.project_key, self.notebook_name, session_id))
 
     def get_state(self):
         """
         Get the status of the notebook
         """
-        if self.state is None:
-            self.state = self.client._perform_json("GET", "/projects/%s/notebooks/" % self.project_key, params={'notebookName' : self.notebook_name})
+        notebook_list = self.client._perform_json("GET",
+                                                 "/projects/%s/jupyter-notebooks/" % self.project_key,
+                                                 params={"active": False})
+        for notebook in notebook_list:
+            if notebook.get("name") == self.notebook_name:
+                    self.state = notebook
```

Comment:
You can assign and return directly there instead of finishing the loop.
Is it expected to not cache the state anymore and always retrieve it?

Reply:
Indeed for the return
For the caching, it's because the state might not have up to date data for `unload`/`get_sessions`. 
For example if we do a `get_notebook` then start a session and try to unload, it will say that the notebook is not running because the sessions in `state` are cached from when there was no sessions.. 
So I don't know what's the best solution :
* Keep the cache but the user must `get_notebook` and `get_state` anyway to have the correct sessions ?
* Keep the changes and always retrieve the latest state ? 
* Or add a `refresh` argument to force refresh the state when we unload/get_sessions ?

Reply:
👍  I'm just wondering about the fact that we retrieve all the notebooks just to have an updated state (We may miss a endpoint to retrieve the state of just one notebook)

Reply:
Yes, I was wondering the same when designing the public API.
I guess we could add an endpoint `GET /project/<projectkey>/notebook/notebookname/sessions` to retrieve the session list

Reply:
Should we do a PR to add this endpoint ?

Reply:
>Should we do a PR to add this endpoint ?

Yes